crates/minimal-oci-spec/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+    bundle_path: &Path,
+) -> Result<PathBuf> {
+    let annotations = build_annotations(image_config);
+    let process = build_process(image_config);
+
+    let spec = SpecBuilder::default()
+        .version(String::from("1.0"))
+        .process(process)
+        .annotations(annotations)
+        .build()?;
+
+    let bundle_config = bundle_path.join(BUNDLE_CONFIG);
+    spec.save(&bundle_config)?;
+
+    Ok(bundle_config)
+}
+
+/// Build process from an image configuration and return it
+fn build_process(image_config: &ImageConfiguration) -> Process {
+    let mut args: Vec<String> = vec![];
+    let mut process = Process::default();
+
+    if let Some(config) = image_config.config() {
+        if let Some(entrypoint) = config.entrypoint() {
+            args.extend(entrypoint.clone());
+        }
+        if let Some(cmd) = config.cmd() {
+            args.extend(cmd.clone());
+        }
+        if let Some(env) = config.env() {
+            process.set_env(Some(env.to_vec()));
+        }
+        if let Some(working_dir) = config.working_dir() {
+            process.set_cwd(PathBuf::from(working_dir));
+        }
+        if !args.is_empty() {
+            process.set_args(Some(args));
+        }
+    }
+    process
+}
+
+/// Build annotations from an image configuration and return it
+fn build_annotations(image_config: &ImageConfiguration) -> HashMap<String, String> {
+    let mut annotations: HashMap<String, String> = HashMap::new();
+    annotations.insert(ANNOTATION_OS.to_string(), image_config.os().to_string());
+    annotations.insert(
+        ANNOTATION_ARCHITECTURE.to_string(),
+        image_config.architecture().to_string(),
+    );
+
+    if let Some(created) = image_config.created() {
+        annotations.insert(ANNOTATION_CREATED.to_string(), created.to_string());
+    }
+    if let Some(author) = image_config.author() {
+        annotations.insert(ANNOTATION_AUTHOR.to_string(), author.to_string());
+    }
+    if let Some(config) = image_config.config() {
+        if let Some(labels) = config.labels() {
+            annotations.extend(labels.clone());
+        }
+        if let Some(stop_signal) = config.stop_signal() {
+            annotations.insert(ANNOTATION_STOP_SIGNAL.to_string(), stop_signal.to_string());
+        }
+        if let Some(exposed_ports) = config.exposed_ports() {
+            annotations.insert(
+                ANNOTATION_EXPOSED_PORTS.to_string(),
+                exposed_ports.join(","),
+            );
+        }
+    }
+    annotations
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile;
+
+    #[test]
+    fn test_bundle_create_config() -> Result<()> {
+        let image_config = ImageConfiguration::default();
+
+        let tempdir = tempfile::tempdir()?;
+        let filename = tempdir.path().join(BUNDLE_CONFIG);
+
+        assert!(!filename.exists());
+
+        assert!(generate_runtime_config(&image_config, tempdir.path()).is_ok());
+        assert!(filename.exists());
```

Comment:
It's good to check that the file is created, but we should at least have another unit test for checking some of its values and compare them with the default ones. They should match.